modules/mobile/app/helpers/mobile/letters_filter.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Mobile
+  class LettersFilter
+    TYPES = %w[
+      benefit_summary
+      benefit_verification
+      civil_service
+      commissary
+      medicare_partd
+      minimum_essential_coverage
+      proof_of_service
+      service_verification
+    ].freeze
+
+    def filter(list)
+      list.select do |letter|
+        if letter.is_a?(EVSS::Letters::Letter)
```

Comment:
Back in December I made a Mobile letter model so we could phase out the EVSS type letter since it'll be removed come August. May be worth using that model so we don't have to do it when other teams decide to rip this one out.

Reply:
EVSS::Letters::Letter is still currently live though right? I had tests fail because I wasn't accounting for that model type

Reply:
yeah it's still in use now. but the mobile letter model should be more or less identical.

Reply:
@aherzberg are you suggesting switching the EVSS letters over to the mobile model? Is the benefit there that we wouldn't have to do this class check? I think we should avoid this class check by moving this model creation line from the letters serializer into the controller:
`Mobile::V0::Letter.new(name: letter[:name], letter_type: letter[:letterType])`
Doing that would mean that everything here is a letter model that responds to the letter_type method. It would also allow us to get rid of the `lighthouse_letters_serializer` method in the letters serializer and run the same code in the serializer for both evss and lighthouse letters. 

The other thing the letters serializer does is change certain values:
```
          letters = letters.map! do |letter|
            letter.name = 'Benefit Summary and Service Verification Letter' if letter.letter_type == 'benefit_summary'
            if letter.letter_type == 'benefit_summary_dependent'
              letter.name = 'Dependent Benefit Summary and Service Verification Letter'
            end
            letter
          end
```
I think we could move that logic into this file and rename this file an adapter instead of a filter. IMO that would be cleaner than updating/creating the model in the serializer.

In the code snippet I quoted above, we have specific handling for benefit_summary_dependent, which is one of the two types that are filtered out. It seems odd that we have specific handling for something that we don't want to display at all anymore. TBH this makes me think this decision to filter them out may have been rushed a bit. Are we sure this feature was fully thought out? It was driven by QA and not by product.

I'm not sure I'm communicating this very clearly, so feel free to reach out to me to talk it through.

Reply:
We want to only allow certain letters through, but in a way that makes it easy to enable them if it's decided we want to support it

Reply:
That is ultimate the desire @aherzberg. Any letter not on our "supported" list should be filtered out, but if it's decided a certain letter is to be supported later then we need to be able to easily enable it to be passed through